### DISCUSSION TITLE
Scale a Quad from center in all directions

### DISCUSSION THREAD

**Discussion by AnweshGangula** [pymupdf/PyMuPDF#1090] [Looking for help] [answered]
I'm searching for text in a pdf and extracting a quad and adding a polygon_annot around it.
But I would like to scale the polygon_annot. How can I do that?

Below is my code:

```python
    for inst in text_instances:
        inst =  inst.transform(fitz.Matrix(2, 2))
        print(inst)
        print(inst.rect)
        # re-ordering the points in list counter-clockwise
        inst[2], inst[3] = inst[3], inst[2]
        highlight = page.add_polygon_annot(inst)
```
i'm currently scaling it with `inst.transform(fitz.Matrix(2, 2))`, but this is to simply multiply the values. How do I scale the values frfom the center of the quad?

**Comment by JorjMcKie** [comment]
You can simply first add the annot.
Then enlarge its ``rect`` by the desired factor using ``annot.set_rect(new_rect)``. There is no need to use a matrix here.
You can either ue a constant value of say 5 pixels in each direction: ``new_rect = annot.rect + (-5, -5, 5, 5)`` or use a factor, e.g. 2:
``new_rect = annot.rect + (-annot.rect.width/2, -annot.rect.height/2, annot.rect.width/2, annot.rect.height/2)``.

**Reply by AnweshGangula** [reply]
I tried this, but I also have text which are rotated ((not horizontal or vertical). For these instances, the rectangle that I'm getting is the smallest containing rectangle, but I want a rectangle that is aligned to text's rotation. That's why I'm using `add_polygon_annot`.

How can I extract the rotation of a search_text?

**Comment by JorjMcKie** [comment]
Manipulating the annot's rect also needs no use of ``annot.update()``

**Comment by JorjMcKie** [comment]
> How can I extract the rotation of a search_text?

When you do `page.search_for(text, quads=True)`` you get a list of quads. A quad with points ``(ul, ur, ll, lr)`` carries the text rotation information in those points. If you use `page.add_highlight_annot(quad)`, then you will already get a rectangle-like background coloring.
Suppose you have this:
![grafik](https://user-images.githubusercontent.com/8290722/122010563-377bb580-cd89-11eb-8fb4-44e849a32dcf.png)
and search like this:
```python
rl = page.search_for("pymupdf", quads=True)
# you will get a quad
>>> quad=rl[0]
>>> page.add_highlight_annot(quad)
'Highlight' annotation on page 0 of x.pdf
>>> doc.ez_save("y.pdf")
```
which results in this:
![grafik](https://user-images.githubusercontent.com/8290722/122010902-917c7b00-cd89-11eb-8b6c-1f5d8a60ec92.png)
You can no problem enlarge this annotation in the way discussed:
![grafik](https://user-images.githubusercontent.com/8290722/122012377-19af5000-cd8b-11eb-873b-fbefb7db45d6.png)
If you happen to dislike highlight annotations, you can do the same thing with a polygon annot created like this:
``annot=page.add_polygon_annot((quad.ul, quad.ur,quad.lr, quad.ll))``.

If you really need the text rotation angle for whatever reason, you can compute the matrix which maps the line from `quad.ul` to `quad.ur` to the x-axis:

```python
>>> mat = fitz.planish_line(quad.ul, quad.ur)
>>> mat
Matrix(0.8660264015197754, 0.4999982416629791, -0.4999982416629791,
0.8660264015197754, 20.10912322998047, -236.43589782714844)
>>>
```
And as you know that a rotation matrix has the form `Matrix(cos, sin, -sin, cos, 0, 0)``, you can easily compute the rotation angle.

**Reply by JorjMcKie** [reply]
Compare that matrix with this one: `fitz.Matrix(30)`!

**Comment by AnweshGangula** [comment, marked as answer]
Thank you for your response @JorjMcKie. But I found that you can use `morph` to scale the **quad**

Below is the function that i'm using to scale the morph by its center:

```python
def getPolygon(inst, scale=1):
    [sumX, sumY] = [0, 0]
    for point in inst:
        sumX += point[0]
        sumY += point[1]
    avgX = sumX/4
    avgY = sumY/4

    # Scale the rectange using Quad.morph(): https://pymupdf.readthedocs.io/en/latest/quad.html#Quad.morph
    inst = inst.morph(fitz.Point(avgX,  avgY), fitz.Matrix(scale, scale))
    # re-ordering the points in list counter-clockwise
    inst[2], inst[3] = inst[3], inst[2]
    return inst
```
